docs/Changelog.md

```diff
@@ -45,6 +45,7 @@
 * [#6826](https://github.com/TouK/nussknacker/pull/6826) Security fix: added validation of expression used inside 
   indexer for Maps and Lists (for example `{1,2,3}[#otherList.remove(1) == null ? 0 : 0]`). This allowed executing 
   some types of unallowed expressions.
+* [#6880](https://github.com/TouK/nussknacker/pull/6880) Performance optimization of generating Avro messages with unions
```

Comment:
Maybe let's add `- shorter message in logs`? :) It ook me a while to find out where is the optimisation ;)

Reply:
Added, along with an explanation in code